File: audit_logger.js
// config/security/audit_logger.js
import fs from 'fs';

const AUDIT_LOG_PATH = process.env.AUDIT_LOG_PATH || './audit.log';

export function auditLog(action, details) {
  const logEntry = {
    timestamp: new Date().toISOString(),
    action,
    details,
  };
  fs.appendFileSync(AUDIT_LOG_PATH, JSON.stringify(logEntry) + '\n');
  console.log('AUDIT:', JSON.stringify(logEntry));
}

File: python/app/audit_logger.py
# config/security/audit_logger.py
import json, os, datetime

AUDIT_LOG_PATH = os.getenv("AUDIT_LOG_PATH", "./audit.log")

def audit_log(action: str, details: dict):
    entry = {
        "timestamp": datetime.datetime.utcnow().isoformat(),
        "action": action,
        "details": details,
    }
    with open(AUDIT_LOG_PATH, "a") as f:
        f.write(json.dumps(entry) + "\n")
    print("AUDIT:", json.dumps(entry))